src/main/java/com/project/Investment/App/generatingModule/GeneratorPosition.java

```diff
@@ -0,0 +1,199 @@
+        List<Position> positions = new ArrayList<>();
+        Map<Integer, Integer> securityIdAndAggregateID = generatorSecurityIdAndAggregateID();
+        Double[] StartBmv = getDistributionBmv(securityIdAndAggregateID, total);
+        LocalDate finishDate = startDate.plusMonths(3);
+
+        for (int i = 0, securityId = 1; i < securityIdAndAggregateID.size(); i++, securityId++) {
+
+            Double bmvGross = StartBmv[i];
+            Double emvGross = getEmvGross(bmvGross);
+            Position position = Position.builder()
+                    .positionId(new PositionId(
+                            entityId,
+                            startDate,
+                            securityId,
+                            securityIdAndAggregateID.get(securityId)))
+                    .frequency('D')
+                    .weight(getWeight(bmvGross, (double) total))
+                    .bmvGross(bmvGross)
+                    .emvGross(emvGross)
+                    .grossReturn(getReturnGross(bmvGross, emvGross))
+                    .gainLossGross(getGainLossGross(emvGross, bmvGross))
+                    .build();
+            positions.add(position);
+        }
+
+        for (LocalDate date = startDate.plusDays(1); date.isBefore(finishDate); date = date.plusDays(1)) {
+
+            if (date.getDayOfWeek().getValue() < 6) {
+
+                int firstPosition = positions.size() - securityIdAndAggregateID.size();
+                Double yesterdayTotal = getYesterdayTotal(positions, firstPosition);
+
+                for (int i = 0, securityId = 1, k = firstPosition; i < securityIdAndAggregateID.size(); i++, securityId++, k++) {
+
+                    Double bmvGross = positions.get(k).getEmvGross();
+                    Double emvGross = getEmvGross(bmvGross);
+
+                    Position position = Position.builder()
+                            .positionId(new PositionId(
+                                    entityId,
+                                    date,
+                                    securityId,
+                                    securityIdAndAggregateID.get(securityId)))
+                            .frequency('D')
+                            .weight(getWeight(bmvGross, yesterdayTotal))
+                            .bmvGross(bmvGross)
+                            .emvGross(emvGross)
+                            .grossReturn(getReturnGross(bmvGross, emvGross))
+                            .gainLossGross(getGainLossGross(emvGross, bmvGross))
+                            .build();
+                    positions.add(position);
+                }
+            }
+        }
+        log.info("Method: generatorPosition - {} Position found", positions.size());
+        return positions;
+    }
+
+    public Map<Integer, Integer> generatorSecurityIdAndAggregateID() {
+        Map<Integer, Integer> securityIdAndAggregateID = new HashMap<>();
+        securityIdAndAggregateID.put(1, 3);
```

Comment:
Try rewriting it with for loop instead of hardcoding

Reply:
Done